packages/e2e/lib/common.dart

```diff
@@ -33,16 +37,21 @@ class Response {
   String toJson() => json.encode(<String, dynamic>{
         'result': allTestsPassed.toString(),
         'failureDetails': _failureDetailsAsString(),
+        if (data != null) 'data': data
       });
 
   /// Deserializes the result from JSON.
   static Response fromJson(String source) {
-    Map<String, dynamic> responseJson = json.decode(source);
-    if (responseJson['result'] == 'true') {
-      return Response.allTestsPassed();
+    final Map<String, dynamic> responseJson = json.decode(source);
+    final String result = responseJson['result'] as String;
+    final dynamic detail = responseJson['failureDetails'];
```

Comment:
Nit: the `detail` seems unnecessary as it's used only once below.

Reply:
Done